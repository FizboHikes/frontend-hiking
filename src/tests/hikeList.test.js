import React from 'react';
import HikeList from '../components/hikeList.js';
import renderer from 'react-test-renderer';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

const userHikes = ["hike1", "hike2"]

test('renders the HikeList correctly', () => {
  const json = renderer
                  .create(<HikeList userHikes={userHikes}/>)
                  .toJSON();
  expect(json).toMatchSnapshot();
})


test('the greeting should display if it is a prop', () => {
  const wrapper = shallow(<HikeList userHikes={userHikes} greeting="hello"/>);
  console.log(wrapper)
  const hikeListShallow = wrapper.find('.hikeList')
  expect(hikeListShallow.find('h2')).toHaveLength(1)
})
