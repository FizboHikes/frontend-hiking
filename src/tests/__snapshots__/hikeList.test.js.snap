// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the HikeList correctly 1`] = `
<div
  className="hikeList"
>
  
  <div
    className="hikeCard"
    style={
      Object {
        "backgroundImage": "url(https://i.pinimg.com/originals/77/85/91/7785910e63b1662e8abe313c8ef9d160.jpg)",
        "backgroundRepeat": "no-repeat",
        "backgroundSize": "cover",
      }
    }
  >
    <div
      className="overlay"
    >
      <div
        className="hikeCardText"
      >
        <a
          href="/hikes/undefined"
        >
            
          <h3>
            "
            "
          </h3>
          <p />
          <p />
        </a>
        <span
          onClick={[Function]}
        >
          ðŸ—‘
        </span>
      </div>
    </div>
  </div>
  <div
    className="hikeCard"
    style={
      Object {
        "backgroundImage": "url(https://i.pinimg.com/originals/77/85/91/7785910e63b1662e8abe313c8ef9d160.jpg)",
        "backgroundRepeat": "no-repeat",
        "backgroundSize": "cover",
      }
    }
  >
    <div
      className="overlay"
    >
      <div
        className="hikeCardText"
      >
        <a
          href="/hikes/undefined"
        >
            
          <h3>
            "
            "
          </h3>
          <p />
          <p />
        </a>
        <span
          onClick={[Function]}
        >
          ðŸ—‘
        </span>
      </div>
    </div>
  </div>
</div>
`;
